### ISSUE TITLE
Model SAE collapse again

### ISSUE THREAD

**Issue by Vincentzyx** [iperov/DeepFaceLab#248]
## Expected behavior

I'm training model SAE using the prebuild windows app (Version: 2019.4.24 DeepFaceLabCUDA10.1AVX) with the configurations as follow:
===== Model summary =====
== Model name: SAE
== Current iteration: 87551
== Model options:
== |== write_preview_history : True
== |== batch_size : 12
== |== sort_by_yaw : False
== |== random_flip : True
== |== resolution : 128
== |== face_type : f
== |== learn_mask : True
== |== optimizer_mode : 2
== |== archi : df
== |== ae_dims : 512
== |== e_ch_dims : 42
== |== d_ch_dims : 21
== |== multiscale_decoder : False
== |== ca_weights : True
== |== pixel_loss : True
== |== face_style_power : 0.0
== |== bg_style_power : 0.0
== |== remove_gray_border : False
== Running on:
== |== [0 : GeForce GTX 1060 6GB]

## Actual behavior

The loss goes high suddenly and the preview image goes red(src) and black(dst), and can't get down anymore.
[14:45:14][#088303][2010ms][2.9678][12.7739]
![PreviewWindow](https://user-images.githubusercontent.com/8458250/56845915-f0a57a00-68fa-11e9-9bb2-f3794fe2c21a.png)
I can provide my model file if it's needed, thank you!

## Steps to reproduce

Don't know the cause clearly, but it seems that when the iteration count goes high, it's more likely to collapse.

## Other relevant information
Command: trainSAE.bat
Software: The prebuild windows app (Version: 2019.4.24 DeepFaceLabCUDA10.1AVX) 
Operating System: Windows 10

**Comment by Greywolf22** [comment]
First of all DeepFaceLabCUDA10.1AVX is for RTX and higher cards. You should use 9.2. 
ca_weights is unstable in my opinion, i would stay away from it

**Comment by Vincentzyx** [comment]
@Greywolf22 Well, the original version of SAE model is easy to collapse too(In my case, when around 10w iterations), don't know the reason…

**Comment by Vincentzyx** [comment]
I don't know much about machine learning, but maybe consider using some sort of optimizer to adjust the learning rate? Or using the clipnorm or clipvalue parameters in Keras to cut the gradient(Gradient Clipping), cut the gradient if it's to big, keep the model away from collapsing.

**Comment by Greywolf22** [comment]
I dont encounter crashes as long as I keep pixel lost and ca weights disabled. Both are enabled in your case

**Comment by tfis2** [comment]
I NEVER had as good results with SAE than with the "old" LIAEF and I tried it a lot.
It has often been told, that SAE contains the old LIAEF Model, but I never got it to work properly.
Especially teeth.

**Comment by iperov** [comment]
Did you read changelog?

```
changed help message for pixel loss:
Pixel loss may help to enhance fine details and stabilize face color. Use it only if quality does not improve over time.
```

you enabled pixel loss too early that cause peaks and collapses
![firefox_2019-04-28_09-31-56](https://user-images.githubusercontent.com/8076202/56859200-7cb9af00-6998-11e9-8c7a-d179c1fe862b.jpg)

**Comment by iperov** [comment]
> ca_weights is unstable in my opinion, i would stay away from it

you are wrong
